src/EFCore.PG/Migrations/NpgsqlMigrationsSqlGenerator.cs

```diff
@@ -708,7 +709,7 @@ public virtual void Generate(NpgsqlDropDatabaseOperation operation, [CanBeNull]
             [NotNull] IModel model,
             [NotNull] MigrationCommandListBuilder builder)
         {
-            var schema = GetSchemaOrDefault(enumType.Schema, model);
+            var schema = enumType.Schema ?? model.Relational().DefaultSchema;
```

Comment:
I think that this logic should actually be inside `PostgresEnum.Schema`, follow the [sequence example](https://github.com/aspnet/EntityFrameworkCore/blob/6e69cad3903233f07e91ab717c195c9ee0644214/src/EFCore.Relational/Metadata/Internal/Sequence.cs#L128) as much as possible.

Reply:
I would prefer that too, but the `PostgresEnum._annotatable` will not always have the default schema set, even where it is logically "active".

That said, this logic is actually present in the getter for `PostgresEnum.Schema`:

```c#
public string Schema => Deserialize(Annotation).Schema ?? (string)_annotatable[RelationalAnnotationNames.DefaultSchema];
```

However, that logic is insufficient for migrations since we forward the enum annotation to `AlterDatabaseOperation`, but [we do not want to forward](https://github.com/npgsql/Npgsql.EntityFrameworkCore.PostgreSQL/pull/685#discussion_r236064018) the `DefaultSchema` annotation.

So in this specific case, `enumType._annotatable` is the `AlterDatabaseOperation`, not the `model`. So calling `enumType.Schema` would return `null` even with the `Sequence`-like getter logic.

Reply:
Ok, will do my best to take another last look in the next few hours so we can lock this down...

Reply:
OK, I see your point. You're right that this logic cannot work inside the annotation.

In that case, should we not be removing the null-coalescing logic from the annotation, and simply return null from its Schema property? It feels wrong to have both.

Reply:
Good point, plus it's probably safer to remove it so there's no ambiguity over whether the schema is set or not.